### DISCUSSION TITLE
Extracted component woes: Can we use @entangle?

### DISCUSSION THREAD

**Discussion by stuartcusackie** [alpinejs/alpine#1312] [4. General]
I like to extract all of my Alpine components to separate files - It's just a personal preference, I really don't like the look of all that inline JS in my html / blade templates. It's great that Alpine allows me to do this but I find the documentation on this is a little lacking, and I'm not sure if some things are possible. For example, Can we entangle livewire data when using extracted components? If so, how?

I'm sure I have more issues with extracted components but can't think of them right now. 

Thanks for reading!

**Comment by ryangjchandler** [comment]
@stuartcusackie Something that I, along with others, do is pass an object through to the function.

```html
<div x-data="myFunction({
    value: @entangle('prop').defer
})">

</div>
```

Then in your JS, you'd have something like this:

```js
function myFunction(config) {
    return {
        value: config.value
    }
}
```

This hides a lot of the logic and you only need to define what you **need**.

**Reply by stuartcusackie** [reply]
Nice one! I'll give this a go. 

Thank you.
